lib/puppet/type/concat_fragment.rb

```diff
@@ -38,11 +42,22 @@
     desc "Tag name to be used by concat to collect all concat_fragments by tag name"
   end
 
+  autorequire(:file) do
+    unless catalog.resource("Concat_file[#{self[:target]}]")
+      warning "Target Concat_file[#{self[:target]}] not found in the catalog"
+    end
+  end
+
   validate do
+    # Check if target is set
+    fail Puppet::ParseError, "Target not set" if self[:target].nil?
+
     # Check if either source or content is set. raise error if none is set
     fail Puppet::ParseError, "Set either 'source' or 'content'" if self[:source].nil? && self[:content].nil?
 
     # Check if both are set, if so rais error
     fail Puppet::ParseError, "Can't use 'source' and 'content' at the same time" if !self[:source].nil? && !self[:content].nil?
+
+    self[:tag] = regsubst(self[:target], '[/:\n\s]', '_', 'GM') if self[:tag].nil?
```

Comment:
You're right. forgot to remove that bit. Need to add a check in the validation block too.